Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -1414,6 +1414,36 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     return .visitChildren
   }
 
+  public override func visit(_ node: PatternBindingSyntax) -> SyntaxVisitorContinueKind {
+    if shouldSkip(node) {
+      return .skipChildren
+    }
+
+    if let identifierPattern = node.pattern.as(IdentifierPatternSyntax.self),
+      let unexpectedBeforeColon = node.typeAnnotation?.unexpectedBeforeColon,
+      let token = unexpectedBeforeColon.onlyToken(where: { $0.tokenKind.isIdentifier })
+    {
+      addDiagnostic(
+        unexpectedBeforeColon,
+        SpaceSeparatedIdentifiersError(
```

Comment:
Do you know why we need to do special handling here and why this is not caught by the general logic in `visit(UnexpectedNodesSyntax)`?

Reply:
The reason was that `previousToken.parent?.is(DeclSyntax.self) == true`. Parent here is an `IdentifierPatternSyntax`. 
I added the pattern as I think it's okay to merge an identifier with the previous identifier pattern.